### PR TITLE
chore: automate releases with release-please

### PR DESCRIPTION
**Pull request by SgtPooki**
This PR enables release-please. some details at https://github.com/ipfs/ipfs-desktop/pull/2671#issuecomment-1760467403

### CODE REVIEW

**Comment by SgtPooki** [comment]
flow: 
1. merge PRs... as normal.. an open release-please PRs will be kept up to date by release-please
2. release-please creates a release PR that collects all these changes on main - https://github.com/ipfs/ipfs-companion/blob/89901efb6d89dcde7b653d8eb6ee379e69c37411/.github/workflows/ci.yml#L96-L108
	1. add files to handle version updates automatically (README.md)
4. ready to release, you merge the release PR - https://github.com/ipfs/ipfs-companion/blob/89901efb6d89dcde7b653d8eb6ee379e69c37411/.github/workflows/ci.yml#L110-L115 runs
	1. git tag *should be created* but currently requires a manual tag creation on ipfs-companion
	2. need to attach binaries via action like https://github.com/ipfs/ipfs-companion/blob/89901efb6d89dcde7b653d8eb6ee379e69c37411/.github/workflows/ci.yml#L149C1-L157C118
		* can also use https://github.com/google-github-actions/release-please-action#attaching-files-to-the-github-release


For testing:
* use option `last-release-sha` to specify a SPECIFIC commit
* use option `release-as` to tag first release as a specific version, and then others will

**Comment by SgtPooki** [comment]
some related testing that was done:

* https://github.com/ipfs/ipfs-desktop/pull/2666
* https://github.com/ipfs/ipfs-desktop/pull/2667
* https://github.com/ipfs/ipfs-desktop/pull/2668
* https://github.com/ipfs/ipfs-desktop/pull/2670
* https://github.com/ipfs/ipfs-desktop/pull/2671
* https://github.com/ipfs/ipfs-desktop/pull/2672

with final documented at https://github.com/ipfs/ipfs-desktop/pull/2672#issuecomment-1760555137

File: README.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
```
Comment (SgtPooki) [inline README.md:2]: the changes in this file are just changing repo strings from old ipfs-shipyard ones: `s/ipfs-shipyard\/ipfs-desktop/ipfs\/ipfs-desktop/`

**Review by SgtPooki — COMMENTED** [review]
self review. I think we're good to go, only took a whole day.

**Review by whizzzkid — COMMENTED** [review]
can be cleaned up more, not blocking though.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -201,6 +201,18 @@ jobs:
...
+      # will attach binaries to that release draft
+      - name: Run release-please github-release
+        if: contains(github.ref, 'refs/tags/') && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
+        uses: google-github-actions/release-please-action@v3.7.3
+        with:
+          command: github-release
+          release-type: node
+          changelog-notes-type: github
+          draft-pull-request: true
+          draft: true
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .github/workflows/ci.yml:214]: the action accepts both parameters, and the one `github-release` is supposed to accept didn't actually create a draft release in https://github.com/ipfs/ipfs-desktop/pull/2672#issuecomment-1760555137 so I decided to keep both. I can remove them, but i'm not sure if we can add artifacts to published github releases, so messing up here can mean updates break for a cycle which would be really painful for users.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -201,6 +201,18 @@ jobs:
         shell: bash
         continue-on-error: true # empty steps.tag.outputs.tag will inform the next step
 
+      # This step will only handle the creation of the release draft, the "Build binaries with electron-builder" step
+      # will attach binaries to that release draft
+      - name: Run release-please github-release
+        if: contains(github.ref, 'refs/tags/') && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .github/workflows/ci.yml:207]: This is within `build` and `test` is already checked inside this one.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -242,3 +254,19 @@ jobs:
...
       - name: Show Cache
         run: du -sh ${{ github.workspace }}/.cache/ && ls -l ${{ github.workspace }}/.cache/
+
+  # This Job will handle the creation of the "release-please" PR
+  create-release-please-pr:
+    runs-on: ubuntu-latest
+    needs: [test, build]
+    if: github.ref == 'refs/heads/main' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
+    steps:
+      - uses: google-github-actions/release-please-action@v3.7.3
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .github/workflows/ci.yml:264]: the plugin is in the PR creation phase and already validated in https://github.com/ipfs/ipfs-desktop/pull/2672

File: release-please-config.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  "bootstrap-sha": "ef61c998865ee90731e0bd495cc825a43c6fac60",
+  "group-pull-request-title-pattern": "chore${scope}: release${component} v${version}",
+  "extra-files": [],
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply release-please-config.json:4]: extending extra-files to do what we needed to do would require changing the inner logic of their primary updater: https://github.com/googleapis/release-please/blob/main/src/updaters/generic.ts#L103

File: release-please-config.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "include-component-in-tag": false,
+  "bump-minor-pre-major": true,
+  "bump-patch-for-minor-pre-major": true,
+  "release-type": "node",
+  "draft": false,
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply release-please-config.json:9]: BAH! that's why the github-release didn't publish as a draft maybe?!

File: package.json [hunk 7]
```diff
@@ -20,7 +20,9 @@
     "build:webui": "run-s build:webui:*",
     "build:webui:download": "npx ipfs-or-gateway -c bafybeiamycmd52xvg6k3nzr6z3n33de6a2teyhquhj4kspdtnvetnkrfim -p assets/webui/ -t 360000 --verbose -g \"https://dweb.link\" ",
     "build:webui:minimize": "shx rm -rf assets/webui/static/js/*.map && shx rm -rf assets/webui/static/css/*.map",
-    "package": "shx rm -rf dist/ && run-s build && electron-builder --publish onTag"
+    "package": "shx rm -rf dist/ && run-s build && electron-builder --publish onTag",
+    "release-pr": "release-please release-pr --token=$GH_TOKEN --plugin '@ipfs-shipyard/release-please-ipfs-plugin' --repo-url=ipfs/ipfs-desktop --trace --debug --draft-pull-request",
+    "release-gh": "release-please github-release --token=$GH_TOKEN --repo-url=ipfs/ipfs-desktop --trace --debug --draft"
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply package.json:25]: I think so.. having commands users can run easily (see developer-notes.md) is very beneficial

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .github/workflows/ci.yml]: removed! you found the bug in another comment :)

File: .github/workflows/ci.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (SgtPooki) [reply .github/workflows/ci.yml:263]: i'm not sure if the other issues you brought up need looked at.. we could just call `npm run release-pr` or `npm run release-gh` appropriately where we need to?

**Review by whizzzkid — APPROVED** [review]
send it

**Comment by SgtPooki** [comment]
Note that the following commits (pushed directly to main) were required to get things actually working:

* https://github.com/ipfs/ipfs-desktop/commit/dfbb26ca706ebe4562d8a24958ef6dd44c6df3f0
* https://github.com/ipfs/ipfs-desktop/commit/6820ffb2d5e6726dbb1dca4134668e32736de393
* https://github.com/ipfs/ipfs-desktop/commit/f6cea9fad451db0311e1a2fd5060cfb6e394c208

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -201,6 +201,17 @@ jobs:
         shell: bash
         continue-on-error: true # empty steps.tag.outputs.tag will inform the next step
```
Comment (rdgoldenempire-oss) [inline .github/workflows/ci.yml:202]: Want to know if the fish game is real cash out and active

**Pull request merged**